### ISSUE TITLE
V2: duplicate marshal functions in convert file

### ISSUE THREAD

**Issue by Smerom** [goadesign/goa#1860]
## Expected Behavior
Valid go code generated

## Actual Behavior
Duplicate marshal functions are generated in the convert file when a type has both an Extend and ConvertTo.

## Steps to Reproduce the Problem (including complete and simple design if relevant)
Design
```golang
package design

import (
	. "goa.design/goa/http/design"
	. "goa.design/goa/http/dsl"
)

var _ = Service("Example", func() {
	Method("show", func() {
		Payload(StructExample)
		//Result(StructExampleExtension)
		Error("internal_error", ErrorResult)
		HTTP(func() {
			GET("/id")
			Response(StatusOK)
			Response("internal_error", StatusInternalServerError)
		})
	})
	Method("otherMethod", func() {
		Payload(StructExampleExtension)
		//Result(StructExampleExtension)
		Error("internal_error", ErrorResult)
		HTTP(func() {
			GET("/hello")
			Response(StatusOK)
			Response("internal_error", StatusInternalServerError)
		})
	})
})

var ExternalStructExample = Type("ExternalStructExample", func() {
	Attribute("someString", String)
})

var StructExample = Type("StructExample", func() {
	ConvertTo(ExampleStruct{})
	Attribute("theStruct", ExternalStructExample)
})

var StructExampleExtension = Type("StructExampleExtension", func() {
	ConvertTo(ExampleStruct{})
	Extend(StructExample)

	Attribute("id", String)
})
```

Resulting convert file:
```golang
// Code generated by goa v2.0.0-wip, DO NOT EDIT.
//
// Example service type conversion functions
//
// Command:
// $ goa gen {path-to-design}/design -o test

package example

import (
	design "{path-to-design}/design"
)

// ConvertToExampleStruct creates an instance of ExampleStruct initialized from
// t.
func (t *StructExample) ConvertToExampleStruct() *design.ExampleStruct {
	v := &design.ExampleStruct{}
	if t.TheStruct != nil {
		v.TheStruct = marshalExternalStructExampleToExternalStructExt(t.TheStruct)
	}
	return v
}

// ConvertToExampleStruct2 creates an instance of ExampleStruct initialized
// from t.
func (t *StructExampleExtension) ConvertToExampleStruct2() *design.ExampleStruct {
	v := &design.ExampleStruct{}
	if t.ID != nil {
		v.ID = *t.ID
	}
	if t.TheStruct != nil {
		v.TheStruct = marshalExternalStructExampleToExternalStructExt(t.TheStruct)
	}
	return v
}

// marshalExternalStructExampleToExternalStructExt builds a value of type
// *design.ExternalStruct from a value of type *ExternalStructExample.
func marshalExternalStructExampleToExternalStructExt(v *ExternalStructExample) *design.ExternalStruct {
	if v == nil {
		return nil
	}
	res := &design.ExternalStruct{}
	if v.SomeString != nil {
		res.SomeString = *v.SomeString
	}

	return res
}

// marshalExternalStructExampleToExternalStructExt builds a value of type
// *design.ExternalStruct from a value of type *ExternalStructExample.
func marshalExternalStructExampleToExternalStructExt(v *ExternalStructExample) *design.ExternalStruct {
	if v == nil {
		return nil
	}
	res := &design.ExternalStruct{}
	if v.SomeString != nil {
		res.SomeString = *v.SomeString
	}

	return res
}
```

## Specifications

  - goa Version: V2 @ commit f14c160845d974e9cf0a9a01abbca41b2adb9483
  - Platform (Linux/Windows etc.): MacOS 10.13

**Comment by Smerom** [comment]
Not sure where to start looking for this one

**Comment by raphael** [comment]
Thank you for the detailed report, this is fixed in #1921
